sunpy/instr/goes.py

```diff
@@ -161,11 +174,12 @@ def temp_em(goeslc, abundances="coronal", download=False, download_dir=DATA_PATH
 
     # Enter results into new version of GOES LightCurve Object
     goeslc_new = copy.deepcopy(goeslc)
-    goeslc_new.data["temperature"] = temp
-    goeslc_new.data["em"] = em
+    goeslc_new.data["temperature"] = temp.value
+    goeslc_new.data["em"] = em.value
```

Comment:
Yes they are.  This is unfortunately necessary at the moment as LightCurve objects cannot contain unit information.  A great improvement to SunPy would be alter LightCurves so that they can.